# From https://github.com/libssh2/libssh2/commit/35ac220a4436df4ce50e990a02721369c9779ea9
From 35ac220a4436df4ce50e990a02721369c9779ea9 Mon Sep 17 00:00:00 2001
From: Will Cosgrove <will@panic.com>
Date: Wed, 26 May 2021 16:42:38 -0700
Subject: [PATCH] configure.ac: don't undefine scoped variable (#594)

* configure.ac: don't undefine scoped variable

To get this script to run with Autoconf 2.71 on macOS I had to remove the undefine of the backend for loop variable. It seems scoped to the for loop and also isn't referenced later in the script so it seems OK to remove it.

* configure.ac: remove cygwin specific CFLAGS #598

Notes:
Remove cygwin specific Win32 CFLAGS and treat the build like a posix build

Credit:
Will Cosgrove, Brian Inglis
---
 configure.ac | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index c9696eccac..c4fc3e4e30 100644
--- a/configure.ac
+++ b/configure.ac
@@ -36,12 +36,9 @@ case "$host" in
     CFLAGS="$CFLAGS -DLIBSSH2_WIN32"
     LIBS="$LIBS -lws2_32"
     ;;
-	*-cygwin)
-	CFLAGS="$CFLAGS -DLIBSSH2_WIN32"
+    *darwin*)
+    CFLAGS="$CFLAGS -DLIBSSH2_DARWIN"
     ;;
-	*darwin*)
-	CFLAGS="$CFLAGS -DLIBSSH2_DARWIN"
-	;;
     *hpux*)
     ;;
     *osf*)
@@ -128,8 +125,6 @@ fi
 m4_set_foreach([crypto_backends], [backend],
   [AM_CONDITIONAL(m4_toupper(backend), test "$found_crypto" = "backend")]
 )
-m4_undefine([backend])
-
 
 # libz
 

